src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityAssemblyLine.java

```diff
@@ -60,4 +74,128 @@ public ICubeRenderer getBaseTexture(IMultiblockPart sourcePart) {
+        }
+    }
+
+    @Override
+    public void onRemoval() {
+        super.onRemoval();
+        if (getWorld().isRemote && beamParticles != null) {
+            for (GTLaserBeamParticle[] particle : beamParticles) {
+                if (particle[0] != null) {
+                    particle[0].setExpired();
+                    particle[1].setExpired();
+                }
+            }
+        }
+        beamParticles = null;
+    }
+
+    private void writeParticles(PacketBuffer buf) {
+        buf.writeVarInt(beamCount);
+    }
+
+    private void readParticles(PacketBuffer buf) {
+        beamCount = buf.readVarInt();
+        if (beamParticles == null) {
+            beamParticles = new GTLaserBeamParticle[17][2];
+        }
+        BlockPos.MutableBlockPos pos = new BlockPos.MutableBlockPos(getPos());
+        for (int i = 0; i < beamParticles.length; i++) {
+            GTLaserBeamParticle particle = beamParticles[i][0];
+            if (i < beamCount && particle == null) {
+                pos.setPos(getPos());
+                Vector3 startPos = new Vector3().add(
+                                pos.move(getFrontFacing().rotateY().getOpposite(), i))
+                        .add(0.5, 0, 0.5);
+                Vector3 endPos = startPos.copy().subtract(0, 1, 0);
+
+                beamParticles[i][0] = new GTLaserBeamParticle(getWorld(), startPos, endPos)
+                        .setBody(laserLocation)
+                        .setBeamHeight(0.125f)
+                        // Try commenting or adjusting on the next four lines to see what happens
+                        .setDoubleVertical(true)
+                        .setHead(laserHeadLocation)
+                        .setHeadWidth(0.1f)
+                        .setEmit(0.2f);
+                beamParticles[i][0].setOnUpdate(p -> { // remove it if machine is inValid
+                    if (!isValid() || getWorld().getTileEntity(getPos()) != this.getHolder()) p.setExpired();
+                });
+                pos.setPos(getPos());
+                startPos = new Vector3().add(
+                                pos.move(getFrontFacing().rotateY().getOpposite(), i).move(getFrontFacing().getOpposite(), 2))
+                        .add(0.5, 0, 0.5);
+                endPos = startPos.copy().subtract(0, 1, 0);
+                beamParticles[i][1] = new GTLaserBeamParticle(getWorld(), startPos, endPos)
+                        .setBody(laserLocation)
+                        .setBeamHeight(0.125f)
+                        // Try commenting or adjusting on the next four lines to see what happens
+                        .setDoubleVertical(true)
+                        .setHead(laserHeadLocation)
+                        .setHeadWidth(0.1f)
+                        .setEmit(0.15f);
+                beamParticles[i][1].setOnUpdate(p -> { // remove it if machine is inValid
+                    if (!isValid() || !GTUtility.isPosChunkLoaded(getWorld(), getPos())) p.setExpired();
```

Comment:
Is there a reason this particle's `setOnUpdate()` has different contents from the one above?